addons/crm/models/utm.py

```diff
@@ -6,37 +6,25 @@
 class UtmCampaign(models.Model):
     _inherit = 'utm.campaign'
 
-    crm_lead_activated = fields.Boolean('Use Leads', compute='_compute_crm_lead_activated')
-    lead_count = fields.Integer('Lead Count', groups='sales_team.group_sale_salesman', compute="_compute_global_opportunity_and_lead_count")
-    opportunity_count = fields.Integer('Opportunity Count', groups='sales_team.group_sale_salesman', compute="_compute_global_opportunity_and_lead_count")
+    use_leads = fields.Boolean('Use Leads', compute='_compute_use_leads')
+    leads_opportunities_count = fields.Integer('Leads/Opportunities count', groups='sales_team.group_sale_salesman', compute="_compute_leads_opportunities_count")
 
-    def _compute_crm_lead_activated(self):
+    def _compute_use_leads(self):
         for campaign in self:
-            campaign.crm_lead_activated = self.env.user.has_group('crm.group_use_lead')
+            campaign.use_leads = self.env.user.has_group('crm.group_use_lead')
 
-    def _compute_global_opportunity_and_lead_count(self):
+    def _compute_leads_opportunities_count(self):
         lead_data = self.env['crm.lead'].with_context(active_test=False).read_group([
             ('campaign_id', 'in', self.ids)],
             ['campaign_id'], ['campaign_id'])
-        data_map = {datum['campaign_id'][0]: datum['campaign_id_count'] for datum in lead_data}
-        if self.env.user.has_group('crm.group_use_lead'):
-            for campaign in self:
-                campaign.lead_count = data_map.get(campaign.id, 0)
-                campaign.opportunity_count = 0
-        else:
-            for campaign in self:
-                campaign.lead_count = 0
-                campaign.opportunity_count = data_map.get(campaign.id, 0)
-
-    def action_redirect_to_leads(self):
-        action = self.env.ref('crm.crm_lead_all_leads').read()[0]
-        action['domain'] = [('campaign_id', '=', self.id)]
-        action['context'] = {'default_type': 'lead', 'active_test': False}
-        return action
+        mapped_data = {datum['campaign_id'][0]: datum['campaign_id_count'] for datum in lead_data}
+        for campaign in self:
+            campaign.leads_opportunities_count = mapped_data.get(campaign.id, 0)
 
-    def action_redirect_to_opportunities(self):
-        action = self.env.ref('crm.crm_lead_opportunities').read()[0]
+    def action_redirect_to_leads_opportunities(self):
+        view = 'crm.crm_lead_all_leads' if self.use_leads else 'crm.crm_lead_opportunities'
+        action = self.env.ref(view).read()[0]
         action['view_mode'] = 'tree,kanban,graph,pivot,form,calendar'
-        action['domain'] = [('campaign_id', '=', self.id)]
-        action['context'] = {'active_test': False}
+        action['domain'] = [('campaign_id', 'in', self.ids)]
+        action['context'] = {'active_test': False, 'create': False}
```

Comment:
Why specifically create False here ?

Reply:
At first it was a copy/paste mistake but it ended up being an LNA approved change. Life is indeed like a chocolate box after all